builtin/providers/consul/config.go

```diff
@@ -26,9 +30,21 @@ func (c *Config) Client() (*consulapi.Client, error) {
 	if c.Scheme != "" {
 		config.Scheme = c.Scheme
 	}
+
+	tlsConfig := &consulapi.TLSConfig{}
+	tlsConfig.CAFile = c.CAFile
+	tlsConfig.CertFile = c.CertFile
+	tlsConfig.KeyFile = c.KeyFile
+	cc, err := consulapi.SetupTLSConfig(tlsConfig)
```

Comment:
`consul/api.TLSConfig{CAFile,CertFile,KeyFile}` are all `string`. Their default values are `""`, so checking to see if `terraform/consul.TLSConfig.{CAFile,CertFile,KeyFile}` (also `string`) are `""` before assigning is unnecessary.

(Answer to your other question below.)